### ISSUE TITLE
Puzzling property in docs for Data.HashMap.Strict.traverseWithKey

### ISSUE THREAD

**Issue by sjakobi** [haskell-unordered-containers/unordered-containers#276]
https://github.com/haskell-unordered-containers/unordered-containers/blob/964edca40d415c7f0b70c99fda2a4d9dc0949b9f/Data/HashMap/Strict/Base.hs#L540-L555

I'm a bit puzzled by the property in the code block. `fmap (map id)` shouldn't make a big difference, so I suspect this is a strictness property?

If this simply means that values are evaluated to WHNF, I think it's maybe a bit redundant, since we already have this info in the module header:

https://github.com/haskell-unordered-containers/unordered-containers/blob/964edca40d415c7f0b70c99fda2a4d9dc0949b9f/Data/HashMap/Strict.hs#L108-L113

**Comment by treeowl** [comment]
I prefer to keep our equations precise, even if it may be a bit annoying.

**Comment by sjakobi** [comment]
So it is a strictness property? That's my main question – sorry for being unclear!

**Comment by treeowl** [comment]
Yes, exactly. It's not really the strictness property we *want*, but it's the strictness property we can *get* with only `Applicative`. `Monad` would let us force values as they're produced.

**Comment by sjakobi** [comment]
Interesting!

What would the strictness property look like if we could use `Monad` instead of `Applicative`?
